src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityCleanroom.java

```diff
@@ -0,0 +1,533 @@
+    }
+
+    private void resetTileAbilities() {
+        this.energyContainer = new EnergyContainerList(new ArrayList<>());
+    }
+
+    @Override
+    protected void formStructure(PatternMatchContext context) {
+        super.formStructure(context);
+        initializeAbilities();
+    }
+
+    @Override
+    public void invalidateStructure() {
+        super.invalidateStructure();
+        resetTileAbilities();
+        this.isClean = false;
+    }
+
+    @Override
+    protected void updateFormedValid() {
+        this.cleanroomLogic.performDrilling();
+        if (!getWorld().isRemote && this.cleanroomLogic.wasActiveAndNeedsUpdate()) {
+            this.cleanroomLogic.setWasActiveAndNeedsUpdate(false);
+            this.cleanroomLogic.setActive(false);
+        }
+    }
+
+    @Override
+    protected BlockPattern createStructurePattern() {
+        // these can sometimes get set to 0 when loading the game, breaking JEI
+        if (width == 0)
+            width = MIN_DIAMETER;
+        if (height == 0)
+            height = MIN_DIAMETER;
+        if (depth == 0)
+            depth = MIN_DIAMETER;
+
+        // build each row of the structure
+        StringBuilder border = new StringBuilder("B"); //      BBBBB
+        StringBuilder wall = new StringBuilder("B"); //        BXXXB
+        StringBuilder inside = new StringBuilder("X"); //      X   X
+        StringBuilder roof = new StringBuilder("B"); //        BFFFB
+        StringBuilder controller = new StringBuilder("B"); //  BFSFB
+
+        // start with block after left edge, do not include right edge with -1
+        for (int i = 1; i < width - 1; i++) {
+            border.append("B");
+            wall.append("X");
+            inside.append(" ");
+            roof.append("F");
+            if (i == width / 2) controller.append("S"); // controller is always centered
+            else controller.append("F");
+        }
+        border.append("B");
+        wall.append("B");
+        inside.append("X");
+        roof.append("B");
+        controller.append("B");
+
+
+        // build each slice of the structure
+        String B = border.toString();
+        String W = wall.toString();
+        String I = inside.toString();
+        String R = roof.toString();
+        String C = controller.toString();
+
+        String[] frontBack = new String[height];
+        String[] inner = new String[height];
+        String[] center = new String[height];
+
+        // bottom and top
+        frontBack[0] = B;
+        frontBack[height - 1] = B;
+        inner[0] = B;
+        inner[height - 1] = R;
+        center[0] = B;
+        center[height - 1] = C;
+
+        // central sections
+        for (int i = 1; i < height - 1; i++) {
+            frontBack[i] = W;
+            inner[i] = I;
+            center[i] = I;
+        }
+
+        TraceabilityPredicate casing = states(getCasingState()).setMinGlobalLimited(width * height * depth * 3 / 5)
```

Comment:
```suggestion
        TraceabilityPredicate casing = states(getCasingState()).setMinGlobalLimited((width * height * depth - (width - 2) * (height - 2) * (depth - 2)) * 3 / 5)
```

I believe its min limit was designed to be 60% of the **wall**, instead of the whole space it takes including the hollow part, or things won't be possible.

Reply:
It is supposed to be the entire volume of the cleanroom, not just the walls. Doing this keeps the amount of required plascrete at an amount that does not allow mass usage of hatches instead of the intended casings. 

Additionally, this is intended to be applied to exclusively the plascrete and not the glass. It cannot be feasibly changed either due to the nature of the multiblock system.